compaction/compaction_manager.cc

```diff
@@ -1185,6 +1185,33 @@ future<> compaction_manager::perform_sstable_scrub_validate_mode(replica::table*
     return perform_task(seastar::make_shared<validate_sstables_compaction_task>(*this, t, std::move(all_sstables)));
 }
 
+future<> compaction_manager::maybe_wait_on_current_compaction(replica::table* t) {
```

Comment:
"Add function to wait for current compaction." is just a means that patch uses to accomplish a goal, which is to not create unbounded number of sstables.